tools/osv-scanner/cmd/osv-scanner/config.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"log"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+)
+
+type Config struct {
+	IgnoredVulnIds []string
+}
+
+// TryLoadConfig tries to load config in `target` or any of it's parent dirs
+// `target` will be the key for the entry in configMap
+// Will shortcut and return "" if globalConfig is not nil
+func TryLoadConfig(target string, configMap map[string]Config) string {
+	if globalConfig != nil {
+		return ""
+	}
+	stat, err := os.Stat(target)
+	if err != nil {
+		log.Fatalf("Failed to stat path: %s", err)
+	}
+
+	if stat.IsDir() && !strings.HasSuffix(target, "/") {
+		// Make sure directories ends with '/'
+		target += "/"
+	}
+
+	currentDir := target
+	for currentDir != "/" {
```

Comment:
I'm wondering if walking the tree to the root could result in a situation where another party with write access to an intermediate directory could insert a broader, maliciously configuration with the malicious intent of hiding a vulnerability? I may be overthinking things, and I suspect there's no specific threat model for what environments this scanner is intended to be run in...

Reply:
Furthermore, now that I've made it to main.go, it looks like this is intended to be run on a Git repo, so realistically walking past the Git repo root shouldn't be necessary, right?

Reply:
That's a good point, I originally had it going all the way up to the root because I thought someone might want to have some global configuration of osv-scanner in their home directory or similar, but that would be pretty niche and better accommodated by having some specific paths set up for where the config files will be located, rather than walking up to root. 

I'll change the behavior to scan only up to the directory initially asked to be scanned, or if no directory has been specified, only a lockfile, just look at the immediate directory that the lockfile is in.

How osv-scanner should load configs is probably worthy of meeting + design doc to clarify the goals.